src/ManagedCodecs/TargaCodec.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System;
+using System.IO;
+using System.Runtime.CompilerServices;
+
+using SixLabors.ImageSharp;
+using SixLabors.ImageSharp.Advanced;
+using SixLabors.ImageSharp.PixelFormats;
+
+using PhotoSauce.MagicScaler;
+
+namespace PhotoSauce.ManagedCodecs
+{
+	/// <summary>Represents a TGA image file.</summary>
+	public sealed class TargaContainer : IImageContainer, IDisposable
+	{
+		private readonly Image decodedImage;
+
+		private TargaContainer(Image image) => decodedImage = image;
+
+		/// <inheritdoc />
+		public FileFormat ContainerFormat => FileFormat.Bmp;
+
+		/// <inheritdoc />
+		public int FrameCount => 1;
+
+		/// <inheritdoc />
+		public IImageFrame GetFrame(int index) => index == 0 ? new TargaFrame(decodedImage) : throw new IndexOutOfRangeException("Invalid frame index");
+
+		/// <summary>Loads a TGA image from the specified path.</summary>
+		/// <param name="imgPath">The path to the image file.</param>
+		/// <returns>A <see cref="TargaContainer"/> encapsulating the image.</returns>
+		public static TargaContainer Load(string imgPath)
+		{
+			int bpp = 0;
+			using (var fs = File.OpenRead(imgPath))
+			{
+				var info = Image.Identify(fs, out var fmt);
+				if (fmt.Name != "TGA")
+					throw new InvalidDataException("Not a TGA image");
+
+				bpp = info.PixelType.BitsPerPixel;
```

Comment:
Would be nice if the `PixelType` was more descriptive of the actual format.  I'm considering some different ways of defining pixel formats in a generic and interoperable way.

Reply:
This is a very good question. An `enum` is no-go, since we are extensible. Could be `System.Type` for `typeof(TPixel)` which is just ... dumb.

Any suggestion? What could be a good general semantical model of a pixel type? /cc @UkooLabs

Reply:
I have something basic set up here: https://github.com/saucecontrol/PhotoSauce/blob/master/src/MagicScaler/Core/PixelFormats.cs

based on: https://docs.microsoft.com/en-us/windows/win32/api/wincodec/nn-wincodec-iwicpixelformatinfo2

I brought over everything except the `ColorContext` (basically a default ICC profile for a given format -- which I don't think makes a ton of sense), and the channel masks (which would be useful for extensibility but are kind of incompatible with performance-oriented scenarios).

It would be great to have a common set of primitives that include PixelTypes in addition to the basic `Rectangle`, `Size`, `Color`, etc that the BCL defines but poorly and that most imaging/graphics libraries end up redefining. See https://github.com/dotnet/runtime/issues/32418

I've been using the GUIDs defined by WIC as a bridge so far, but that doesn't seem very sustainable.  If we could all agree on a registry for GUIDs, it seems we could agree on an actual structure to represent the formats.

Reply:
I'm not familiar enough with pixel types in general, but I like the approach you take! 

Two critical points however:
- `PixelColorRepresentation` prevents extensibility. One can easily imagine pixel formats not fitting into it.
- `PixelValueEncoding`: I'm not aware of any pixel format in ImageSharp, that is not `Linear`

Are these values important for your use cases?

Reply:
> I'm not aware of any pixel format in ImageSharp, that is not Linear

My understanding is that almost every `PixelValueEncoding` in ImageSharp is `Companded`.  One can generally assume that any format with less than 16 bits per channel is companded, because linear in 8-bit just doesn't work.

In a 16-bit+ format, it could be either, or it could be something wacky like [scRGB](https://en.wikipedia.org/wiki/ScRGB).  I'm not too concerned about scRGB, because it seems nobody other than Microsoft ever adopted it.  But if you give me a pixel value in 32 bit per channel floating point, it's very important to know whether the value is gamma adjusted (companded) or not before operating on it.  This information could be provided in the form of a color profile (which could be even more useful), but there's some question of whether a profile would belong to the pixel format or to the source of the pixel data (I keep it on the source).  The distinction between `Companded` vs `Linear` is enough to know whether a certain operation (e.g. compositing) will work properly with the values.

`PixelColorRepresentation` is a tricky one.  My handling of that so far it is to have common representations like RGB or CMYK defined, but to treat any that have specialized use (like Cb or Cr planar formats) as `Unspecified`.  That value basically allows any operation that doesn't require knowledge of the format to still work on the pixels (e.g. convolution), where something where channel meaning or order is important would need an exact match.

But none of that is fully fleshed out in my mind, which is why I've kept that part of the API internal for now.  You guys have been far more brave in sharing public APIs :)

Reply:
@saucecontrol What do the `PixelAlphaRepresentation` values mean?

Reply:
Associated/Unassociated is just alternate terminology for Premultiplied/PresentButNotPremultiplied.  I tend to think of Premultiplied as being in terms of the color channels and Associated as describing the alpha channel itself.